types/store/index.ts

```diff
@@ -47,7 +47,7 @@ export type GlobalState = BaseGlobalState & {
             };
             focusedPostId: string;
             mobileView: boolean;
-            keepChannelIdAsUnread: Channel | null;
+            lastUnreadChannel: (Channel & {hadMentions: boolean}) | null; // Actually only an object with {id: string, hadMentions: boolean}
```

Comment:
```suggestion
            lastUnreadChannel: (Pick<Channel, 'id'> & {hadMentions: boolean}) | null; // Actually only an object with {id: string, hadMentions: boolean}
```
If it's just the `id`?

Reply:
I tried changing the type on this to match the actual type, but it ran into some other issues because everything else expects `lastUnreadChannel` to be a full Channel, including some code in mattermost-redux, and I wanted to avoid having to make a bunch of changes to both repos to clean that up. We can address that more easily once they're eventually in the same repo.